client/src/partials/budget/new/new_budget.js

```diff
@@ -0,0 +1,357 @@
+  session.no_data_msg = null;
+
+  // Basic setup function when the models are loaded
+  function startup(models) {
+    angular.extend($scope, models);
+
+    session.found = models.budgets.data.length > 0;
+    session.numPeriods = models.budgets.data.length;
+    var total = 0.0;
+    models.budgets.data.forEach(function (b) {
+      b.freeze = false;
+      b.editing = false;
+      total += Number(b.budget);
+    });
+    session.totalBudget = total;
+
+    if (session.found) {
+      recompute();
+    }
+    else {
+      session.no_data_msg = $translate.instant('BUDGET.EDIT.DATA_NOT_FOUND')
+        .replace('(acct)', session.account.account_txt)
+        .replace('(fyname)', session.fiscal_year.fiscal_year_txt);  
+    }
+  }
+
+  // Initialize editing the selected account
+  function submitAccount(newAccount) {
+    if (newAccount) { 
+      session.account = newAccount;
+      dependencies.account.query.where = ['account.id=' + newAccount.id];
+      dependencies.budgets.query.where = ['period.fiscal_year_id=' + session.fiscal_year.id, 'AND',
+        'budget.account_id=' + session.account.id, 'AND',
+        'period.fiscal_year_id=' + session.fiscal_year.id, 'AND',
+        'period.period_number<>0'];
+      //  NOTE: Restricting the periods to the selected fiscal year
+      //  automatically limits the budget items to ones for this
+      //  enterprise since the specific FY is tied to a particular
+      //  enterprise.
+      dependencies.periods.query.where = ['period.fiscal_year_id=' + session.fiscal_year.id, 'AND',
+      'period.period_number<>0'];
+      validate.refresh(dependencies, ['account', 'budgets', 'periods'])
+      .then(startup);
+      session.mode = 'edit';
+    }
+  }
+
+  function accountWhere() {
+    // only unlocked income/expense accounts
+    // Allow accounts of class 1 and 2 used in budget
+    return [['account.account_type_id IN (1,4)','OR',['account.classe IN (1,2,5)','AND','account.is_used_budget=1']], 'AND', 'account.locked=0' ];
+  }
+
+  function restartSearch() {
+    session.mode = 'configure';
+    session.option = 'manual';
+    session.autoAdjust = false;
+  }
+
+  function createBudget() {
+    var newBudgets = [];
+    $scope.periods.data.forEach(function (per, index) {
+    newBudgets.push({
+      'account_id' : session.account.id,
+      'period_id'  : per.id,
+      'budget'     : 0.0});
+    });
+
+    connect.post('budget', newBudgets)
+    .then(function () {
+      messenger.success($translate.instant('BUDGET.EDIT.CREATE_OK'));
+      submitAccount(session.account);
+    })
+    .catch(function (err) {
+    messenger.danger($translate.instant('BUDGET.EDIT.CREATE_FAIL'));
+    console.log(err);
+    });
+  }
+
+  function updateBudget() {
+    // Save the budget data for all the periods
+    $http.post('/budget/update', { 
+      params : {
+        budgets   : $scope.budgets.data,
+        accountId : session.account.id
+      }
+    })
+    .success(function () {
```

Comment:
The $http promise methods `.success()` and `.error()` are deprecated.  Do not use them.